frontend/src/views/Room.jsx

```diff
@@ -110,21 +115,67 @@ function Room() {
     }
   };
 
+  const handleRoleChange = (payload) => {
+    if (payload.eventType === 'INSERT') {
+      const { id, userEmail } = payload;
+      const permission = payload['rooms-permission'].name;
+      dispatch(addUpdateParticipant({
+        name: userEmail,
+        role: permission,
+        id,
+      }));
+    }
+    // Supabase realtime only sends the ID that was deleted from the rooms-data table
+    if (payload.eventType === 'DELETE') {
+      const { id } = payload.old;
+      dispatch(removeRole({ id }));
+    }
+  };
+
   // initialize room
   useEffect(() => {
+    const updateParticipantRoles = async () => {
+      const initialParticipantRoles = await getRoomPermissions(roomId);
+      initialParticipantRoles.map((part) => dispatch(addUpdateParticipant({
+        name: part.userEmail,
+        role: part['rooms-permission'].name,
+        id: part.id,
+      })));
+    };
     const subscribeToRemoteStreams = async (r) => {
       // subscribe ta all remote participants for testing purposes
       const { remoteParticipants } = r;
       const rps = Array.from(remoteParticipants.values());
       await Promise.all(rps.map(async (rp) => {
-        await rp.subscribe();
-      }));
+        dispatch(addUpdateParticipant({
+          name: rp.id,
+          role: ROLES.GUEST,
+        }));
+        rp.subscribe();
+        // Add remote participants to participants list.
+      })).then(() => {
+        updateParticipantRoles();
+      });
```

Comment:
Also, you used `await` before, so you shouldn't need `.then ()` here.

```suggestion
      }));
      updateParticipantRoles();

```